### ISSUE TITLE
osgEarthAnnotation's FeatureNode build twice

### ISSUE THREAD

**Issue by youngerking1985** [gwaldron/osgearth#738]
setMapNode will call build(), setStyle also will call build()

FeatureNode::FeatureNode(MapNode* mapNode,
                         Feature* feature,
                         const Style& in_style,
                         const GeometryCompilerOptions& options,
                         StyleSheet* styleSheet) :
AnnotationNode(),
_options           ( options ),
_needsRebuild      ( true ),
_styleSheet        ( styleSheet )
{
    _features.push_back( feature );

    FeatureNode::setMapNode( mapNode );
    
    Style style = in_style;
    if (style.empty() && feature->style().isSet())
    {
        style = *feature->style();
    }

    setStyle( style );
}

**Comment by jasonbeverage** [comment]
Are you seeing an issue with this or are you just noting a potential inefficiency?

**Comment by youngerking1985** [comment]
yes, with large features, my data will hundred thousand vector lines. so build twice will more slower
